File: tests/fixtures/open_redirect/ruby/safe_redirect.rb
# Safe: query arg routed through `validate_redirect_url` allowlist before
# being passed to redirect_to.
class HomeController
  def validate_redirect_url(raw)
    raw.is_a?(String) && raw.start_with?('/') ? raw : '/'
  end

  def jump
    target = params[:next]
    safe = validate_redirect_url(target)
    redirect_to safe
  end
end

File: tests/fixtures/open_redirect/rust/safe_redirect.rs
// Safe: tainted value routed through `validate_redirect_url` allowlist
// before being passed to `Redirect::to`.
use axum::response::Redirect;
use std::env;

fn validate_redirect_url(raw: &str) -> String {
    if raw.starts_with('/') {
        raw.to_string()
    } else {
        "/".to_string()
    }
}

fn bounce() -> Redirect {
    let next = env::var("NEXT").unwrap_or_default();
    let safe = validate_redirect_url(&next);
    Redirect::to(&safe)
}
